### ISSUE TITLE
"built-in referenced when not iterating" is reported even on yield from iterable

### ISSUE THREAD

**Issue by radkujawa** [PyCQA/pylint#2643]
### Steps to reproduce
run pylint on:
```
def f():
    yield from range(1,5)

for i in f():
    print(i)
```

### Current behavior
```[W1638(range-builtin-not-iterating), f] range built-in referenced when not iterating```
Analogical behavior is for `yield from zip`, `yield from map`, etc.

### Expected behavior
no error.

### pylint --version output
```
$ pylint --version
pylint 2.2.2
astroid 2.1.0
Python 3.6.5 (default, May 17 2018, 16:32:59) 
[GCC 5.4.0 20160609]
```
